api/v1beta1/policyautomation_types.go

```diff
@@ -52,6 +54,25 @@ type AutomationDef struct {
 	TowerSecret string `json:"secret"`
 	// JobTTL sets the time to live for the Kubernetes AnsibleJob object after the Ansible job run has finished.
 	JobTTL *int `json:"jobTtl,omitempty"`
+	// The maximum number of violation contexts will be provided to the Ansible Tower as extra variable.
+	// +kubebuilder:validation:Minimum=0
+	PolicyViolationContextLimit uint `json:"policyViolationContextLimit,omitempty"`
+}
+
+// ViolationContext defines the non-compliance replicated policy information
+// that were sent to the Anisbile job through extra_var.
+type ViolationContext struct {
+	// +kubebuilder:validation:Required
```

Comment:
I would remove these `kubebuilder` markers since this struct won't be generated into a CRD.